### ISSUE TITLE
react/jsx-sort-props breaks the code

### ISSUE THREAD

**Issue by davesnx** [yannickcr/eslint-plugin-react#1993]
**Tell us about your environment**

    "babel-eslint": "^9.0.0",
    "eslint": "5.6.0",
    "eslint-plugin-react": "^7.11.1"

* **ESLint Version:** 9.0.0
* **Node Version:** 10.6.0
* **npm Version:** -
* **yarn Version:** 1.9.4

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**Please show your full configuration:**
I created a repo to reproduce the bug: https://github.com/davesnx/eslint-bug-react-sort-props

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parser: 'babel-eslint',
  plugins: ['react'],
  rules: {
    'react/jsx-sort-props': 1
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import React, { Component } from 'react'
import PropTypes from 'prop-types'

const PictureChoice = () => {
  return (
    <List
      className={className}
      onStageAnswer={onStageAnswer}
      onCommitAnswer={onCommitAnswer}
      isFocused={isFocused}
      direction={direction}
      allowMultipleSelection={allowMultipleSelection}
      measureLongestChildNode={measureLongestChildNode}
      layoutItemsSize={layoutItemsSize}
      handleAppScroll={handleAppScroll}
      isActive={isActive}
      resetSelection={resetSelection}
      onKeyboardChoiceHovered={onKeyboardChoiceHovered}
      keyboardShortcutType
    />
  )
}

export default PictureChoice

```

```bash
npx eslint . --fix
```

**What did you expect to happen?**
So theorically the rule should re-order my props, in the worst case scenario, not sort them. But actually is breaking them. It's a weird output:

```js
import React, { Component } from 'react'
import PropTypes from 'prop-types'

const PictureChoice = () => {
  return (
    <List
      allowMultipleSelection={allowMultipleSelection}
      className={className}
      direction={direction}
      isFocused=handleAppScroll={handleAppScroll}ection=isActive={isActive}owMultiplisFocused={isFocused}ardShortcutType
      layoutItemsSize={layoutItemsSl
      layoutItemsSize={layoutItemsSize}
      handleAppScroll={handleAppScroll}
      isActive={isActive}
      resetSelection={resetSelection}
      onKeyboardChoiceHovered={onKeyboardChoiceHovered}
      keyboardShortcutType
    />
  )
}

export default PictureChoice
```

It's actually trying to sort in a weird way, and it's not consistent.
Happens with a boolean props as well

```js

      isFocused=
      
      handleAppScroll={handleAppScroll}
      
      ection=
      
      isActive={isActive}
      
      owMultiplisFocused={isFocused}
      
      ardShortcutType
      layoutItemsSize={layoutItemsSl
      
```

**Comment by ljharb** [comment]
Does this only happen with babel-eslint?

**Comment by alexzherdev** [comment]
I think this is a duplicate of #1922. This rule doesn’t make an attempt of handling multiline props properly :/

**Comment by ljharb** [comment]
In that case we should entirely disable the autofixer, pending making it behave properly.

**Comment by julia-dizhak** [comment]
same happens for me with `autofixer`
